Automattic/wp-calypso
# TypeScript: Move base tsconfig to calypso-build

#### Changes proposed in this Pull Request

TypeScript: Move base tsconfig to calypso-build. This seems to make sense, since the structure is quite similar to how many of our other configs extend their 'base' versions found in `calypso-build`.

Follow-up from #37632

**Not quite working yet, currently falsely taking stuff in `exclude`d dirs into account.**

Docs: https://www.typescriptlang.org/docs/handbook/tsconfig-json.html

#### Testing instructions

No functional changes.

A project specific typecheck can be run:

```sh
npm run typecheck -- --project client/landing/gutenboarding
```

or for Calypso-the-app:

```sh
npm run typecheck
```

[Check the `typecheck` job](https://circleci.com/gh/Automattic/wp-calypso/521728), it should report the same output as current `master`.
(Same if you run the above commands locally, on this branch vs `master`. Note that Gutenboarding will have one more error on this branch because of some import from `client/`. We'll likely squash those when polishing #37945.)
